modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -428,18 +610,19 @@ def _read(cls, path, engine, columns, **kwargs):
                     reason="Mixed partitioning columns in Parquet",
                     **kwargs,
                 )
-        dataset = cls._get_dataset(path, kwargs.get("storage_options") or {})
+
+        dataset = cls.get_dataset(path, engine, kwargs.get("storage_options") or {})
         index_columns = (
-            dataset.schema.pandas_metadata.get("index_columns", [])
-            if dataset.schema.pandas_metadata
+            dataset.pandas_metadata.get("index_columns", [])
+            if dataset.pandas_metadata
             else []
         )
         # If we have columns as None, then we default to reading in all the columns
-        column_names = dataset.schema.names if not columns else columns
+        column_names = dataset.columns if not columns else columns
```

Comment:
```suggestion
        column_names = columns if columns else dataset.columns
```
I personally feel positive conditions read better